restapi/api_client.go

```diff
@@ -134,6 +141,17 @@ func NewAPIClient(opt *apiClientOpt) (*api_client, error) {
 		debug:                 opt.debug,
 	}
 
+	if opt.use_oauth_client_credentials {
```

Comment:
We could eliminate the `use_oath_client_credentials` boolean and just check here for non-default values like so:
```
if opt.oath_client_id != "" && opt.oath_client_secret != "" && opt.oauth_token_url != "" && len(opt.oath_scopes) > 0
```

Reply:
👍